corehq/apps/hqmedia/static/hqmedia/js/hqmedia.reference_controller.js

```diff
@@ -1,349 +1,371 @@
-        //                for (var i = 0; i < res.length; i++) {
-        //                    self.searchOptions.push(new MediaOption(self, res[i]))
-        //                }
-        //            });
-        //        }
-    };
+        // Needed for Upload Controller
 
-    self.triggerUpload = function () {
-        self.upload_controller.resetUploader();
-        self.upload_controller.currentReference = self;
-        if (self.upload_controller) {
-            self.upload_controller.uploadParams = {
-                path: self.path,
-                originalPath: self.path,
-                media_type: self.media_class,
-                old_ref: self.m_id || "",
-                replace_attachment: true,
-            };
-        }
-        self.upload_controller.updateUploadFormUI();
-    };
+        // we don't want to be dependent on a knockout structure (vellum)
+        self.getUrl = function () {
+            return self.url();
+        };
 
-    // Needed for Upload Controller
+        self.isMediaMatched = function () {
+            return self.is_matched();
+        };
 
-    // we don't want to be dependent on a knockout structure (vellum)
-    self.getUrl = function () {
-        return self.url();
+        // bound to event mediaUploadComplete
+        self.uploadComplete = function (trigger, event, data) {
+            if (data && !data.errors.length) {
+                self.setObjReference(data.ref);
+            }
+        };
+    }
+
+    function ImageReference(ref) {
+        'use strict';
+        var self = {};
+        BaseMediaReference.call(self, ref);
+        self.upload_controller = HQMediaUploaders['hqimage'];
+        self.preview_template = "image-preview-template";
+        self.thumb_url = ko.computed(function () {
+            return (self.url()) ? self.url() + "?thumb=50" : "";
+        }, self);
+
+        return self;
+    }
+
+    ImageReference.prototype = Object.create(BaseMediaReference.prototype);
+    ImageReference.prototype.constructor = ImageReference;
+
+
+    function AudioReference(ref) {
+        'use strict';
+        BaseMediaReference.call(this, ref);
+        var self = this;
+        self.upload_controller = HQMediaUploaders['hqaudio'];
+        self.preview_template = "audio-preview-template";
+    }
+
+    AudioReference.prototype = Object.create(BaseMediaReference.prototype);
+    AudioReference.prototype.constructor = AudioReference;
+
+
+    function VideoReference(ref) {
+        'use strict';
+        BaseMediaReference.call(this, ref);
+        var self = this;
+        self.upload_controller = HQMediaUploaders['hqvideo'];
+        self.preview_template = "video-preview-template";
+    }
+
+    VideoReference.prototype = Object.create(BaseMediaReference.prototype);
+    VideoReference.prototype.constructor = VideoReference;
+
+
+    // Kept from Tim, you might want to fix it up a bit
+    function MediaOption(mediaRef, data) {
```

Comment:
Not touching this function for this migration as it looks waiting for a fix

Reply:
Actually, could you just delete this? Looks like this functionality has been commented out since 2013: https://github.com/dimagi/commcare-hq/commit/246deaa2abf926cbc91015054a71c3b030fc2855

I'd leave the search function & error, but delete the comments and MediaOption.